shared_model/validators/query_validator.hpp

```diff
@@ -35,10 +35,13 @@ namespace shared_model {
     template <typename FieldValidator>
     class QueryValidatorVisitor
         : public boost::static_visitor<ReasonsGroupType> {
-     public:
-      QueryValidatorVisitor(const FieldValidator &validator = FieldValidator())
+      QueryValidatorVisitor(const FieldValidator &validator)
           : validator_(validator) {}
 
+     public:
+      QueryValidatorVisitor(std::shared_ptr<ValidatorsConfig> config)
+          : QueryValidatorVisitor(FieldValidator{config}) {}
```

Comment:
```suggestion
          : QueryValidatorVisitor(FieldValidator{std::move(config)}) {}
```
if it is needed there ofc